src/rosdep2/main.py

```diff
@@ -368,6 +368,11 @@ def _rosdep_main(args):
                       help="Affects the 'update' verb. "
                            'If specified end-of-life distros are being '
                            'fetched too.')
+    parser.add_option('-t', '--dependency-types', dest='dependency_types',
+                      type="choice", choices=list(all_dep_types()),
+                      default=[], action='append',
+                      help='Dependency types to install, can be given multiple times. '
+                           'Choose from {}. Default: all except doc.'.format(all_dep_types()))
```

Comment:
This doesn't work as written, you should probably use the following options:

```suggestion
    parser.add_option('-t', '--dependency-types',
                      type=str, choices=list(all_dep_types()),
                      nargs='+',
                      help='Space-separated list of dependency types to install. Default: all except "doc"'
```

* nitpick: `dest` automatically replaces hyphens with underscores, this is a no-op
* `type` should name the type of argument being parsed - `"choice"` is not a type, `str` is: this raises an exception right now because `'choice' is not callable`
*  `nargs='+'` allows one or more arguments, defaulting to an empty list (meaning you don't need to specify `default` or `action`) - without this, if you pass in multiple choices, then the second one will raise `unrecognized argument: buildtool`. If you use `nargs='+', action='append'` in combination, then you'll actually get a nested list `dependency_types=[['build', 'exec']]` rather than the desired `dependency_types=['build', 'exec']`

Reply:
Note the parser is from [optparse](https://docs.python.org/3/library/optparse.html), so `type` is a name from an extendable list of [options](https://docs.python.org/3/library/optparse.html#optparse-standard-option-types) and `nargs` is a fixed [number](https://docs.python.org/3/library/optparse.html#optparse.Option.nargs).

Currently, the user needs to repeat the arg keyword like `-t build -t exec -t test`. However, I agree it would be better to provide the list directly. To do this, we can do like [`--skip-keys`](https://github.com/ros-infrastructure/rosdep/blob/2977726b32f57205ac0985ce9eb8f44e5dbbe8bf/src/rosdep2/main.py#L404-L405) which expect a string with comma-separated names like `--skip-keys "pkg1 pkg2 pkg3"` or, even better, we could create a custom action like this [example](https://docs.python.org/3/library/optparse.html#adding-new-actions) that allows the user to specify comma-separated list  like `--dependency-types build,exec,test`

Reply:
Ah - that's on me for assuming argparse - sorry about that. I just didn't even think of it since it's been deprecated for a while. I  think space-separated is preferable - comma-separated is not a very common commandline paradigm.